src/compiler/transformers/jsx.ts

```diff
@@ -113,8 +113,12 @@ namespace ts {
                     || createAssignHelper(currentSourceFile.externalHelpersModuleName, segments);
             }
 
-            const element = createReactCreateElement(
-                compilerOptions.reactNamespace,
+            const jsxFactory = compilerOptions.jsxFactory
+                ? createJsxFactory(compilerOptions.jsxFactory)

```

Comment:
Since this is a fully-synthesized entity name, consider caching it on `SourceFile`, similar to `externalHelpersModuleName` rather than reparsing it each time it is hit. You could also create it once at the top of `transformJsx`.